ls1intum/Artemis
# [Chore + Mini-Features] Metis exercise/lecture discussion section improvements

### Checklist

### Motivation and Context


The discussion section (former "Postings" former "Q&A") on exercise and lecture pages had some improper translations, such as "Postings" (in German and English), which I wanted to change. Thereby, I realized that many other styling issues should be resolved, the structure in general could be improved to a cleaner version and some mini-features should be added. 
I decided to go for a "Chore+Mini-Features PR" 😄 

### Description

**Styling and Appearance**
- change of colors (too many nested grey boxes, icons too black), sizes of icons
- introduce different icons for post (bubble) and answer post (two bubbles)
- fixed button misalignment in edit mode (cancel and save button)
- re-structuring of answers and posts e.g. line divider between posts

**Translation and labels**
- remove labels and tooltips where not necessary (e.g. post, answer above the content -> label, style and placement should be enough to identify a posts and answers)
- section name is changed to `Discussion`
- German translation for Post is `Beitrag`
- all Metis related translations for the discussion section and overview were factored out in dedicated translation files
- adapt overlooked translation keys for course and exercise statistics

**Refactoring/Renaming notification domain**
The notification component was not refactored yet (server side).

**Mini-Features**
- Input validation: it should not be possible to created or save (after edit) posts or answers without content
- Loading feedback: the save-button was changed to a button with spinner indication during loading time (It happened to me sometimes, that I got impatient when I did not receive any feedback of the loading process and clicked on save two times, which led to duplicated posts)

### Steps for Testing


Note: I did not change the logic of voting or approving, please focus on reporting issues regarding the style or the wording of the discussion section, or bug related to the mini-features mentioned above.

1. Log in to Artemis
2. Navigate to an Exercise or Lecture page and use the discussion section
3. As student: add posts and answers, edit own posts
4. As instructor: add posts and answers (automatically approved), edit posts and answers of anybody
5. Check if the notifications (bell at the top) for the new actions are created properly (with the new terms: post/Beitrag, reply/Antwort)

As instructor:
1. Check course statistics: no missing translation keys for post related statistics
2. Check exercise statistics: no missing translation keys for post related statistics

### Test Coverage




### Screenshots


Before:
<img width="563" alt="Bildschirmfoto 2021-06-13 um 22 17 20" src="https://user-images.githubusercontent.com/41366522/121820863-4bde8600-cc95-11eb-92f3-4efa4fbad665.png">
After:
<img width="514" alt="Bildschirmfoto 2021-06-13 um 22 15 29" src="https://user-images.githubusercontent.com/41366522/121820855-4123f100-cc95-11eb-96ae-7e50447c9ee6.png">
